app/DynamicsAdapter/DynamicsAdapter.Web/Mapping/Converters.cs

```diff
@@ -0,0 +1,113 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using AutoMapper;
+using DynamicsAdapter.Web.PersonSearch.Models;
+using Fams3Adapter.Dynamics.Address;
+using Fams3Adapter.Dynamics.Identifier;
+using Fams3Adapter.Dynamics.OptionSets.Models;
+
+namespace DynamicsAdapter.Web.Mapping
+{
+    public class InformationSourceConverter : IValueConverter<int?, string>
+    {
+        public string Convert(int? source, ResolutionContext context)
+        {
+            return Enumeration.GetAll<InformationSourceType>().FirstOrDefault(m => m.Value == source)?.Name;
+        }
+    }
+
+    public class IssuedByTypeConverter : IValueConverter<string, int?>
+    {
+        public int? Convert(string source, ResolutionContext context)
+        {
+            InformationSourceType sourceType = Enumeration.GetAll<InformationSourceType>().FirstOrDefault(m => m.Name.Equals(source, StringComparison.OrdinalIgnoreCase));
+            return (sourceType == null) ? InformationSourceType.Other.Value : sourceType.Value;
+        }
+    }
+
+    public class IdentifierTypeConverter : IValueConverter<int?, int?>
+    {
+        public int? Convert(int? source, ResolutionContext context)
+        {
+            if (source == IdentificationType.DriverLicense.Value) return (int)PersonalIdentifierType.DriverLicense;
```

Comment:
I think you should have a dictionary and then resolve from it there is so many if else here

Reply:
I do not understand.